test/create_zarr_ss2.py

```diff
@@ -0,0 +1,194 @@
+import argparse
+import csv
+import glob
+import json
+import os, sys
+import re, logging
+
+import crimson.picard
+import numcodecs.blosc
+import numpy
+import zarr
+
+""" key,  dataset name,   suffix of file,  metadata description 
+    this is a list of groups to add to the zarr file, which can be 
+    further extended
+"""
+_FILES = {
+  "QCs" : [ "QCs", "*_QCs.csv",  "QC data" ], 
+  "genes_results"  : [ "TPM", "*_rsem.genes.results", "count for the genes and also isoforms" ],
+}
+
+COMPRESSOR = numcodecs.blosc.Blosc(cname='lz4', clevel=5, shuffle=1, blocksize=0)
+
+def init_zarr(sample_id, path, fileformat):
+    """  Creates an empty zarr file  in the path
+        Args:
+            path:  path to the zarr 
+        Returns:
+            
+        Raises:
+    """
+    if fileformat=="DirectoryStore":
+        store = zarr.DirectoryStore(path)
+
+    if fileformat=="ZipStore":
+       store= zarr.ZipStore(path, mode='w')
+
+    # create the root group
+    root = zarr.group(store, overwrite=True)
+
+    # add some readme for the user
+    root.attrs['README'] = "The schema adopted in this zarr store may undergo changes in future"
+    root.attrs['sample_id'] = sample_id
+
+    # now iterate through list of expected groups and create them
+    for data in _FILES:
+      datagroup = root.create_group(data, overwrite=True)
+      datagroup.attrs['description'] = _FILES[data][2]
+
+    return root
+
+
+def read_and_convert_QC(root, input_files_string):
+    """Converts the QC of Smart Seq2 gene file pipeline outputs to zarr file
+        Args:
+            root: root object for the zarr
+            analysis_output_path: path where the SS2 pipeline are found 
+        Returns:
+            True if success else False
+        Raises:
+
+    """
+    
+    # split the files string by , 
+    # crate an array of the file names containing the suffix _QCs.csv 
+    relevant_qc_files =  [  x.strip() for x in input_files_string.split(',')  if re.search(r'_QCs.csv$', x) ]
+
+    # we expect only one, write an warning 
+    if len(relevant_qc_files) !=1:
+       logging.warning(sys.stderr.write("WARNING: Multiple files ending with _QCs.csv. Expected only one"))
+
+
+    # read the QC values
+    QC_values = []
+    with open(relevant_qc_files[0], 'r') as f:
+       reader = csv.reader(f)
+       for row in reader:
+          QC_values.append(row)
+
+    datagroup = root['QCs']
+
+    # Metrics
+    """ The first value is the row name so we remove it"""
```

Comment:
I'd use an in-line comment with `#` since the comment doesn't need multiple lines.